pkg/apply/task/apply_task.go

```diff
@@ -102,6 +102,9 @@ func (a *ApplyTask) Start(taskContext *taskrunner.TaskContext) {
 			// Set the client and mapping fields on the provided
 			// info so they can be applied to the cluster.
 			info, err := a.InfoHelper.BuildInfo(obj)
+			// BuildInfo strips path annotations.
+			// Use modified object for filters, mutations, and events.
+			obj = info.Object.(*unstructured.Unstructured)
```

Comment:
I agree that making a copy of the input resource to the library before making changes is a better way here.
Reusing the `obj` name here is a bit confusing. Could we maybe write this as:
```golang
for i := range objects {
    info, err := a.InfoHelper.BuildInfo(objects[i])
    obj := info.Object.(*unstructured.Unstructured)
```

Reply:
Reusing the object from the for loop is technically slightly more efficient. It re-uses the variable instead of a new variable per iteration being garbage collected. Not sure the memory or performance impact really matters, but it seems like an arbitrary change. The comment should be enough to avoid confusion.